Sprint-1/implement/sum.test.js

```diff
@@ -34,3 +34,33 @@ test.todo("given an empty array, returns 0")
 // Given an array with only non-number values
 // When passed to the sum function
 // Then it should return the least surprising value given how it behaves for all other inputs
+
+// Test: Empty array
+test("given an empty array, returns 0", () => {
+    expect(sum([])).toBe(0);
+});
+
+// Test: Single number
+test("given an array with one number, returns that number", () => {
+    expect(sum([42])).toBe(42);
+});
+
+// Test: Array with positive and negative numbers
+test("given an array with positive and negative numbers, returns the correct sum", () => {
+    expect(sum([10, -5, 20, -10])).toBe(15);
+});
+
+// Test: Array with decimal numbers
+test("given an array with decimal numbers, returns the correct sum", () => {
+    expect(sum([1.5, 2.5, 3.5])).toBe(7.5);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?